### ISSUE TITLE
Header text with FRAME style type is misplaced in scores

### ISSUE THREAD

**Issue by ingvarskagg** [musescore/MuseScore#22734]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

Can be reproduced on scores with a text in the header (inside the frame) that has “FRAME” style. This text is not a title, subtitle, composer or etc., so its position can overlap other header text elements like lyricist or other FRAME text. 

This is a rare issue to observe in MU Studio, but it cause various layout issues in MU Play and cannot be efficiently fix in XTZ. More context can be found in [Jira](https://mu--se.atlassian.net/browse/XTC-799).

### Steps to reproduce

Open a score with FRAME style text in the header and note the text element's position. Example of the score:
[Disney_Medley_For_Woodwind_Quintet.mscz.zip](https://github.com/musescore/MuseScore/files/15260681/Disney_Medley_For_Woodwind_Quintet.mscz.zip)

### Screenshots/Screen recordings

Score from MU Studio v1.3
<img width="1343" alt="Screenshot 2024-05-09 at 13 51 23" src="https://github.com/musescore/MuseScore/assets/149785325/8a5b0eb3-9035-4c99-99df-939daf9e9a43">

### MuseScore Version

4.2.1

### Regression

No.

### Operating system

macOS 14

**Comment by mike-spa** [comment]
I'm sorry, but it's not possible to solve this issue. I would even say that this is not an issue, it is just the nature of how frames and text work in Musescore.

In Musescore, you can create a frame, and you can insert any number of text elements inside of it. For any text elements that is inside a frame, Musescore allows you to specify how to align it with respect to the frame. This forms part of the "Text style", so a text with "Title" style gets aligned to the top-center
<img width="1223" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/3ddb5a10-da58-466e-8c5f-4f0a53419698">
a text with "Composer" style gets aligned to the bottom-right, etc.
<img width="1226" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/c135805a-a169-4887-a5f8-21cdef7fc5dc">

But anyway, regardless of text styles, we are just working with relative positions. Musescore just positions the text element wherever you ask it to, similarly to what any graphics vector editor would do. In your example, it doesn't matter that the text sytle is set to "Frame". What matters is that the text element is set to be aligned to the top-left, and that is exactly what Musescore is doing. 
<img width="1082" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/9923b0c9-d0ba-420c-8705-9fd5ed8077e8">

Trying to do collision-avoidance between text elements is, as far as all I can think, simply impossible. Or at least I don't know of any graphics editor that can do that.

1. In order to avoid collisions, the program would have to move the text elements. But that contradicts the user setting, because then the element is not anymore aligned to the top-left of the frame.
2. Resolving collisions for a generic number of elements is an unsolvable problem. Also because:
3. There is no way to generally determine what goes above/below what. For example, how do I decide if this is the correct solution
![image](https://github.com/musescore/MuseScore/assets/93707756/a5a63c58-d77d-4c3a-bbf6-79ed94a3cbb0)
or this?
![image](https://github.com/musescore/MuseScore/assets/93707756/969ec481-4edf-4acd-b528-f6465133e8f7)
4. There can be cases, like this, where there is simply not enough space in the frame, so we would have to make the frame larger, but that's also not possible because the height of the frame is set by the user (and it's important that it stays that way because having frames of unpredictable height could be a problem)
<img width="503" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/9178b3c8-10f6-4021-8e2e-22568ef344c9">

So, to summarize, here Musescore is just behaving like a vector graphics editor here. And this kind of things can only be manually solved by the user, I'm afraid.
